apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/ItemController.java

```diff
@@ -67,6 +69,13 @@ public OpenItemDTO getItem(@PathVariable String appId, @PathVariable String env,
     return this.itemOpenApiService.getItem(appId, env, clusterName, namespaceName, key);
   }
 
+  @GetMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/encodeItems/{key:.+}")
+  public OpenItemDTO getItemByEncodeKey(@PathVariable String appId, @PathVariable String env, @PathVariable String clusterName,
+                             @PathVariable String namespaceName, @PathVariable String key) {
+    key = new String(Base64.getUrlDecoder().decode(key.getBytes(StandardCharsets.UTF_8)));
```

Comment:
How about extracting this duplication of `new String(Base64.getUrlDecoder().decode(key.getBytes(StandardCharsets.UTF_8)))` into UrlUtils?

Reply:
that's a good idea.